fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/IndexBuilderImpl.java

```diff
@@ -129,75 +120,60 @@ private void rebuild() {
+            fedoraIds.add(fedoraId);
+            if (headers.isArchivalGroup() || headers.isObjectRoot()) {
+                rootId.set(fedoraId);
+            }
 
-                                if (parentId != null) {
-                                    this.containmentIndex.addContainedBy(txId, parentId,
-                                            headers.getId());
-                                }
+            if (!headers.isDeleted()) {
+                if (!fedoraId.isRepositoryRoot()) {
+                    var parentId = headers.getParent();
 
-                                headersList.add(headers);
-                            }
+                    if (headers.getParent() == null) {
+                        if (headers.isObjectRoot()) {
+                            parentId = FedoraId.getRepositoryRootId();
+                        } else {
+                            throw new IllegalStateException(String.format("Resource %s must have a parent defined",
+                                    fedoraId.getFullId()));
                         }
-                    } catch (PersistentStorageException e) {
-                        throw new RepositoryRuntimeException(format("fedora-to-ocfl index rebuild failed: %s",
-                                e.getMessage()), e);
                     }
-                }
-            });
-
-            // if a resource is not an AG then there should only be a single resource per OCFL object
-            if (fedoraIds.size() == 1 && rootId.get() == null) {
-                rootId.set(fedoraIds.get(0));
-            }
 
-            fedoraIds.forEach(fedoraIdentifier -> {
-                var rootFedoraIdentifier = rootId.get();
-                if (rootFedoraIdentifier == null) {
-                    rootFedoraIdentifier = fedoraIdentifier;
+                    this.containmentIndex.addContainedBy(txId, parentId, fedoraId);
+                    headersList.add(headers.asKernelHeaders());
                 }
-                fedoraToOcflObjectIndex.addMapping(txId, fedoraIdentifier, rootFedoraIdentifier, ocflId);
-                LOGGER.debug("Rebuilt fedora-to-ocfl object index entry for {}", fedoraIdentifier);
-            });
-
-            headersList.forEach(headers -> {
-                searchIndex.addUpdateIndex(txId, headers);
-                LOGGER.debug("Rebuilt searchIndex for {}", headers.getId());
-            });
+            }
+        });
 
-        } catch (final PersistentStorageException e) {
-            throw new RepositoryRuntimeException("Failed to rebuild fedora-to-ocfl index: " +
-                    e.getMessage(), e);
+        // if a resource is not an AG then there should only be a single resource per OCFL object
+        if (fedoraIds.size() == 1 && rootId.get() == null) {
+            rootId.set(fedoraIds.get(0));
         }
+
+        fedoraIds.forEach(fedoraIdentifier -> {
+            var rootFedoraIdentifier = rootId.get();
```

Comment:
Yes, it's a hold over from an original iteration of this class. I changed it to throw an exception if the root id isn't found, because it should always be.

Reply:
It seems like you can remove 
```
            var rootFedoraIdentifier = rootId.get();
            if (rootFedoraIdentifier == null) {
                rootFedoraIdentifier = fedoraIdentifier;
            }
```
since a null root id now throws an exception a few lines above this?